hibernate-envers/src/main/java/org/hibernate/envers/query/internal/impl/RevisionsOfEntityQuery.java

```diff
@@ -71,6 +74,36 @@ public RevisionsOfEntityQuery(
 		this.includePropertyChanges = includePropertyChanges;
 	}
 
+	@Override
+	public AuditAssociationQuery<? extends AuditQuery> traverseRelation(
+			String associationName,
+			JoinType joinType,
+			String alias) {
+
+		if ( !selectEntitiesOnly ) {
+			throw new IllegalStateException( "Audit association queries only permitted while selecting only entities." );
+		}
+
+		AbstractAuditAssociationQuery<AuditQueryImplementor> query = associationQueryMap.get( associationName );
+		if ( query == null ) {
+			query = new RevisionsOfEntityAssociationQuery<>(
+					enversService,
+					versionsReader,
+					this,
+					qb,
+					associationName,
+					joinType,
+					aliasToEntityNameMap,
+					REFERENCED_ENTITY_ALIAS,
+					alias
+			);
+
+			addAssociationQuery( associationName, query );
+		}
```

Comment:
In this case, Map's `putIfAbsent` seems better, as follows:
```
associationQueryMap.putIfAbsent( associationName, name -> new RevisionsOfEntityAssociationQuery<>(
					enversService,
					versionsReader,
					this,
					qb,
					associationName,
					joinType,
					aliasToEntityNameMap,
					REFERENCED_ENTITY_ALIAS,
					alias
			) );
```

Reply:
It looks like ```addAssociationQuery``` adds the subquery to two different datastructures - associationQueryMap and associationQueries (list).

Reply:
You might be able to remove the list and use ```associationQueryMap.values()```